docs/pages/router/reference/api-routes.mdx

```diff
@@ -196,6 +198,163 @@ export async function GET(request: Request, { post }: Record<string, string>) {
+  // This will delay the response:
+  await pingAnalytics(...);
+
+  const data = await fetchExampleData(...);
+  return Response.json({ data });
+}
+```
+
+In the above example, we have an `await`-ed function call that delays the rest of the API route's execution. If we don't want to delay a `Response` then `await`-ing this call isn't suitable. However, calling the function without `await` wouldn't guarantee that this task keeps a serverless function running.
+
+Instead, you can use `expo-server`'s [`runTask()`](/versions/latest/sdk/server/#runtaskfn) helper to run concurrent tasks. This is equivalent to the [`waitUntil()`](https://developer.mozilla.org/en-US/docs/Web/API/ExtendableEvent/waitUntil) method that you see in service worker code or other serverless runtimes.
+
+```ts app/tasks+api.ts
+import { runTask } from 'expo-server';
+
+export async function GET(request: Request) {
+  // This will NOT delay the response:
+  runTask(async () => {
+    await pingAnalytics(...);
+  });
+
+  const data = await fetchExampleData(...);
+  return Response.json({ data });
+}
+```
+
+With `runTask`, you have a compromise between `await`-ing and not `await`-ing asynchronous functions. They'll be run concurrently, and don't delay the API route's response or execution, but are also making sure the runtime is aware of them, and don't quit early.
+
+However, sometimes you might also want to delay a task until after the API route has returned a `Response` and don't want to run it if the API route has rejected or only run a function after a time-sensitive task has completed, to avoid concurrent code from delaying computation heavy tasks in your API route.
+
+You can use `expo-server`'s [`deferTask()`](/versions/latest/sdk/server/#defertaskfn) helper to schedule tasks to run after a `Response` has been resolved by your API route.
+
+```ts app/tasks+api.ts
+import { deferTask } from 'expo-server';
+
+export async function GET(request: Request) {
+  // This will run after this entire function resolves:
+  deferTask(async () => {
+    await pingAnalytics(...);
+  });
+
+  const data = await fetchExampleData(...);
+  return Response.json({ data });
+}
+```
+
+### Response headers
+
+When structuring and splitting server logic into separate helper functions and files, it may be necessary to modify `Response` headers before a `Response` has been created.
+
+For example, you may need to add metadata in [server middleware](/router/reference/middleware/) to a `Response` before your API route code is running.
+
+```ts app/+middleware.ts
+import { setResponseHeaders } from 'expo-server';
+
+export default function middleware(request: Request) {
+  // Rate limiters typically add a `Retry-After` header
+  setResponseHeaders({ 'Retry-After': '3600' });
+}
+```
+
+In the above example, we're adding a `Retry-After` header to a future `Response` that an API route may be creating. This can also be extended for authentication and cookies.
```

Comment:
```suggestion
In the above example, a `Retry-After` header is added to a future `Response` that an API route may be creating. This can also be extended for authentication and cookies.
```